src/Events/StatusChangesParams.js

```diff
@@ -4,45 +4,46 @@ class StatusChangesParams {
    */
   constructor (filter = null) {
     this.filter = filter
+    this.sort = null
   }
 
   /**
    * @returns {StatusChangesParams}
    */
   sortByObjectLabel () {
-    this.sort = 'objectLabel'
+    this.sort = this.sort ? this.sort.replace(/[a-zA-Z]*:/, 'objectLabel:') : 'objectLabel:asc'
```

Comment:
what does this .replace() call do?

Reply:
Oh I see, it replaces everything before the colon by 'objectLabel:'. 

Wouldn't it be cleaner to have 2 fields: `sortField` and `sortDirection`? Then you don't need funky regex'es.

Reply:
I tried a few different approaches, but couldn't figure out how to achieve the same effect. I want to be able to chain parameters and still get {sort, filter} object to send to the server, but if I introduce `sortField` and `sortDirection`, I need a way to combine them and I don't want to have to rely on a helper function to do this. Can you think of a cleaner approach?

Reply:
For instance, I want to be able to simply do this without any middle helper functions: 

```
let params = new StatusChangesParams().sortByDate().sortDescending().sortByObjectLabel().sortAscending()
  let firstPage = await client.events.statusChanges(event.key).firstPage(params)
```

Reply:
Perhaps it is an overkill, I see that the PHP client doesn't have a parameter object at all, so maybe I should do the same...

Reply:
Sure, that would work.

If you prefer to keep the parameter object, you can use 2 fields, `sortField` and `sortColumn`, which are strings. When doing the actual API call you can concatenate the values of those 2 fields.

Reply:
Yes, but there is one Lister and one PageFetcher classes in the JS client, so it should apply to all the objects that use them, and that's why I wanted to handle it in the StatusChangesParams object. I think instead I will introduce a helper class that handles the string replacement and keep the rest as it is if that's okay.

Reply:
What I meant was adding a `serialize` method or something, which converts the Params object to a plain JS object. That way you can decouple the internal structure of the Params object (with 2 fields for sorting) from the structure our API expects (1 field for sorting).

`StatusChangesParams` would become simpler, since you don't need to do any regex replacements.

That being said, there's no single right way of implementing this. Just sharing what is in my eyes easier to understand code.

```
class StatusChangesParams {

  constructor (filter = null) {
    this.filter = filter
    this.sortField = null
    this.sortDirection = 'asc'
  }

  sortByObjectLabel () {
    this.sortField = 'objectLabel'
    return this
  }

  ...

  sortAscending () {
    this.sortDirection = 'asc'
    return this
  }

  sortDescending () {
    this.sortDirection = 'desc'
    return this
  }

  ...

  serialize() {
    return {
      filter: this.filter,
      sort: this.sortField ? this.sortField + ":" + this.sortDirection : null,
    }
  }
}
```